openwisp_controller/config/tests/test_admin.py

```diff
@@ -2160,6 +2098,41 @@ def test_vpn_template_switch(self):
             )
             self.assertEqual(config.vpnclient_set.count(), 1)
 
+    # helper for asserting queries executed during template fetch for a device
+    def _verify_template_queries(self, config, count):
+        path = reverse(f'admin:{self.app_label}_device_change', args=[config.device.pk])
+        for i in range(count):
+            self._create_template(name=f'template-{i}')
+        expected_count = 24
+        if django.VERSION < (5, 2):
+            # In django version < 5.2, there is an extra SAVEPOINT query
+            # leading to extra RELEASE SAVEPOINT query, thus 2 extra queries
+            expected_count += 2
+        with self.assertNumQueries(expected_count):
+            # contains 22 queries for fetching normal device data
+            response = self.client.get(path)
+            # contains 2 queries, 1 for fetching organization
+            # and 1 for fetching templates
+            response = self.client.get(
+                reverse(
+                    'admin:get_relevant_templates', args=[config.device.organization.pk]
+                )
+            )
+            self.assertEqual(response.status_code, 200)
+
+    # ensuring queries are consistent for different number of templates
+    def test_templates_fetch_queries_1(self):
+        config = self._create_config(organization=self._get_org())
+        self._verify_template_queries(config, 1)
+
+    def test_templates_fetch_queries_5(self):
+        config = self._create_config(organization=self._get_org())
+        self._verify_template_queries(config, 1)
+
+    def test_templates_fetch_queries_10(self):
+        config = self._create_config(organization=self._get_org())
+        self._verify_template_queries(config, 1)
+
```

Comment:
@DragnEmperor can you help me understand wht do we need three tests?

Reply:
I have added these checks to ensure that queries are not changed and remain the same irrespective of number of templates

Reply:
The count argument is `1` in all the tests. Is that correct?